pkg/providers/instance/offerings/resperrorhandlers.go

```diff
@@ -0,0 +1,94 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package offerings
+
+import (
+	"context"
+	"errors"
+
+	sdkerrors "github.com/Azure/azure-sdk-for-go-extensions/pkg/errors"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/karpenter-provider-azure/pkg/cache"
+	"github.com/Azure/skewer"
+	corecloudprovider "sigs.k8s.io/karpenter/pkg/cloudprovider"
+)
+
+type responseErrorHandler struct {
+	match  func(error) bool
+	handle commonErrorHandle
+}
+
+type ResponseErrorHandling struct {
+	UnavailableOfferings  *cache.UnavailableOfferings
+	ResponseErrorHandlers []responseErrorHandler
+}
+
+func (h *ResponseErrorHandling) extractErrorCodeAndMessage(err error) (string, string) {
+	var respErr *azcore.ResponseError
+	if errors.As(err, &respErr) {
+		return respErr.ErrorCode, respErr.Error()
+	}
+	return "", err.Error()
+}
+
+func (h *ResponseErrorHandling) HandleResponseError(ctx context.Context, sku *skewer.SKU, instanceType *corecloudprovider.InstanceType, zone, capacityType string, responseError error) error {
```

Comment:
minor
```suggestion
func (h *ResponseErrorHandling) HandleError(ctx context.Context, sku *skewer.SKU, instanceType *corecloudprovider.InstanceType, zone, capacityType string, responseError error) error {
```

Or even just Handle